### PR TITLE
CVSB-3630

### PR DESCRIPTION
**Pull request by mdclarke21**
Also fixes CVSB-3748, CVSB-3662 and CVSB-3712

### CODE REVIEW

File: src/providers/utils/common-functions.ts [hunk 1]
```diff
@@ -19,10 +19,13 @@ export class CommonFunctionsService {
 
   // convert date to format DD MMM YYYY (e.g. 21 Feb 2019)
   public convertDate(dateString: string): string {
-    let date: Date = new Date(Date.parse(dateString));
-    return `${date.getDate()} 
-            ${date.toLocaleString('en-us', {month: 'short'})} 
-            ${date.getFullYear()}`;
+    if(dateString){
```
Comment (danieliordache) [inline src/providers/utils/common-functions.ts]: you can use a pipe for it if it's used only for display

File: src/pages/testing/vehicle/vehicle-history-details/vehicle-history-details.ts [hunk 2]
```diff
@@ -43,10 +56,25 @@ export class VehicleHistoryDetailsPage {
     this.selectedTestType = this.testResultHistory[this.testIndex].testTypes[this.testTypeIndex];
     this.testTypeResults = TEST_TYPE_RESULTS;
     this.defaultValues = DEFAULT_VALUES;
+
     this.testsWithoutCertificate = TestsWithoutCertificate.TestsWithoutCertificate;
     this.testsWithoutTestExpiry = TestsWithoutTestExpiry.TestsWithoutTestExpiry;
     this.testsWithoutSeatbelts = TestsWithoutSeatbelts.TestsWithoutSeatbelts;
     this.testsWithoutDefects = TestsWithoutDefects.TestsWithoutDefects;
+    
+    this.doesNotHaveCert = this.commonFunc.checkForMatchInArray(this.selectedTestType.testTypeName, this.testsWithoutCertificate);
```
Comment (danieliordache) [inline src/pages/testing/vehicle/vehicle-history-details/vehicle-history-details.ts]: can you add those lines to a method(ex: formatDataForDisplay) for readability

File: src/pages/testing/vehicle/vehicle-history-details/vehicle-history-details.ts [hunk 3]
```diff
@@ -43,10 +56,25 @@ export class VehicleHistoryDetailsPage {
...
+    this.doesNotHaveDefects = this.commonFunc.checkForMatchInArray(this.selectedTestType.testTypeName, this.testsWithoutDefects);
+    this.doesNotHaveBelts = this.commonFunc.checkForMatchInArray(this.selectedTestType.testTypeName, this.testsWithoutSeatbelts);
+    this.doesNotHaveExpiry = this.commonFunc.checkForMatchInArray(this.selectedTestType.testTypeName, this.testsWithoutTestExpiry);
+    this.isTestResultAbandon = this.commonFunc.checkForMatch(this.selectedTestType.testResult, TEST_TYPE_RESULTS.ABANDONED);
+    this.isTestResultFail = this.commonFunc.checkForMatch(this.selectedTestType.testResult, TEST_TYPE_RESULTS.FAIL);
+    this.testTypeDate = this.commonFunc.convertDate(this.selectedTestType.testTypeStartTimestamp);
+    this.testTypeStartTime = this.commonFunc.getTimeFromDate(this.selectedTestType.testTypeStartTimestamp);
+    this.testTypeEndTime = this.commonFunc.getTimeFromDate(this.selectedTestType.testTypeEndTimestamp);
+    this.testExpiryDate = this.commonFunc.convertDate(this.selectedTestType.testExpiryDate);
```
Comment (danieliordache) [inline src/pages/testing/vehicle/vehicle-history-details/vehicle-history-details.ts]: if it's used only for display, use a pipe (ex: Angular's date pipe)

File: src/models/tests/test-type.model.ts [hunk 4]
```diff
@@ -15,6 +15,8 @@ export interface TestTypeModel {
   reasonForAbandoning: string | null;
   additionalNotesRecorded: string | null;
   additionalCommentsForAbandon: string | null;
+  testNumber?: string;
```
Comment (danieliordache) [inline src/models/tests/test-type.model.ts]: if those are used only for FE purposes, move them under the line :)

File: src/providers/utils/common-functions.ts (hunk 1 above)
Comment (mdclarke21) [reply src/providers/utils/common-functions.ts]: Done

File: src/pages/testing/vehicle/vehicle-history-details/vehicle-history-details.ts (hunk 3 above)
Comment (mdclarke21) [reply src/pages/testing/vehicle/vehicle-history-details/vehicle-history-details.ts]: Done

File: src/models/tests/test-type.model.ts (hunk 4 above)
Comment (mdclarke21) [reply src/models/tests/test-type.model.ts]: Done

File: src/pages/testing/vehicle/vehicle-history-details/vehicle-history-details.ts (hunk 2 above)
Comment (mdclarke21) [reply src/pages/testing/vehicle/vehicle-history-details/vehicle-history-details.ts]: Done

File: src/pipes/format-date/format-date.pipe.ts [hunk 5]
```diff
@@ -0,0 +1,21 @@
+import { Pipe, PipeTransform } from '@angular/core';
+import { DEFAULT_VALUES } from '../../app/app.enums';
+
+@Pipe({
+  name: 'formatDate'
+})
+
+export class FormatDatePipe implements PipeTransform {
```
Comment (danieliordache) [inline src/pipes/format-date/format-date.pipe.ts]: it's not possible to use Angular's build-in date pipe ?

Comment (mdclarke21) [reply src/pipes/format-date/format-date.pipe.ts]: Yes, I'm stupid. Done :)

**Pull request merged**